src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt

```diff
@@ -610,7 +614,8 @@ class CourtSentencingSynchronisationService(
             telemetry,
           )
         } ?: let {
-        dpsApiService.createSentence(nomisSentence.toDpsSentence(event.offenderIdDisplay)).run {
+        // retrieve offence mappings (created as part of the court appearance flow)
+        dpsApiService.createSentence(nomisSentence.toDpsSentence(event.offenderIdDisplay, getDpsChargeMappings(nomisSentence))).run {
           tryToCreateSentenceMapping(
```

Comment:
Main change - when a sentence is created it is also associated with existing offender charges